packages/base-contract/src/utils.ts

```diff
@@ -38,3 +38,30 @@ export function methodAbiToFunctionSignature(methodAbi: MethodAbi): string {
     const method = AbiEncoder.createMethod(methodAbi.name, methodAbi.inputs);
     return method.getSignature();
 }
+
+/**
+ * Replaces unliked library references in bytecode with real addresses
+ * and returns the bytecode.
+ */
+export function linkLibrariesInBytecode(
+    bytecodeArtifact: EvmBytecodeOutput,
+    libraryAddresses: { [libraryName: string]: string },
+): string {
+    let bytecode = bytecodeArtifact.object.substr(2);
+    for (const link of Object.values(bytecodeArtifact.linkReferences)) {
+        for (const [libraryName, libraryRefs] of Object.entries(link)) {
+            const libraryAddress = libraryAddresses[libraryName];
+            if (!libraryAddress) {
+                continue;
```

Comment:
I feel like we should be throwing here instead of in the contract wrapper, ideally with both the name of the contract whose bytecode we're modifying and the name of the missing library.